### PR TITLE
feat(generate, init): allow user to override path for templates

### LINKED ISSUE #28: Allow developers to configure path for templates
## Summary

Right now, we don't expose the `--path` flag when using the `generate` command. By default, the folder/files generated go to `src/components` for the `react-component` flag. The only way for the developer to override this is by adding in their own templates. 

Instead, we should allow them to pass a `--path` flag. We could then use this in the `init` command when we add the script to their `package.json`. 

### Basic example

For the `generate` command:
`eb-scripts generate -t react-component -n MyNewComponent -p src/myCustomFolder`

For the `init` command, we would simply use the flag when we add this script to their `package.json`:
`"g:component": "eb-scripts generate -t react-component -n -p src/components"`

Then we'd add documentation saying you could modify the path in the script under the `-p flag`

### Motivation

Not all projects have the same folder structure. Some may want to generate files elsewhere. We use a default but allow the user to overwrite it.

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR refactors both the `init` and the `generate` so that users can overwrite the default location for generated templates (`src/components`)

## Changes
- refactor flags or args for init, remove `-p` in favor of arg
- update jest to ignore `/lib` and focus on `src`
- add `-p` for `--path` to `generate`
- allow user to pass `-p` to generate command

## Screenshots

image

## Checklist

- [x] tested locally
- [ ] installed new dependencies
- [ ] updated the docs
- [x] added a test

Fixes #28

### CODE REVIEW

File: src/commands/init.ts [hunk 1]
```diff
@@ -31,40 +31,37 @@ type ScriptByProject = {
...
+      description: `The language or framework of the project.\nValid options: ${validProjectTypes}`,
+      options: validProjectTypes
     }
   ];
 
   async run() {
-    const { flags } = this.parse(Init);
-    const project = flags.project || DEFAULT_PROJECT_NAME;
+    const { args } = this.parse(Init);
+    console.log("hello args", args);
```
Comment (cball) [inline src/commands/init.ts]:
```suggestion
```

File: src/commands/init.ts [hunk 2]
```diff
@@ -31,40 +31,37 @@ type ScriptByProject = {
 const pathWhereScriptIsRunning = process.cwd();
 const scriptsByProject: ScriptByProject = {
   react: {
-    "g:component": "eb-scripts generate -t react-component -n"
+    "g:component": "eb-scripts generate react-component -p src/components -n"
   }
 };
 
-const validProjectTypes: ProjectType[] = ["react"];
+// Use the keys from scriptsByProject as valid projec types
```
Comment (cball) [inline src/commands/init.ts]:
```suggestion
// Use the keys from scriptsByProject as valid project types
```

**Comment by jsjoeio** [comment]
🦅👀 Thanks for the review @cball !

**Comment by jsjoeio** [comment]
> 🔥 this is such a great library!

Thank you @cmejet 🙏

**Pull request merged**